torch/_dynamo/exc.py

```diff
@@ -67,7 +69,7 @@ def __init__(self, backend_fn, inner_exception):
 class Unsupported(TorchDynamoException):
     def __init__(self, msg):
         super().__init__(msg)
-        self.real_stack = []
+        self.real_stack = torch._guards.TracingContext.extract_stack()
```

Comment:
Instead of adding stack entries as we walk up the exception frames, we always have an accurate user stack at any given point in time, so we can just use it directly. Making exceptions make use of this facility is not strictly necessary, but stylistically coherent.